# IMPORTING THE REQUIRED LIBRARIES

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
from plotly.offline import iplot,plot
from plotly.subplots import make_subplots
import plotly.io as pio

import warnings
warnings.filterwarnings("ignore")

# LOADING THE DATASET

In [2]:
df=pd.read_csv("Sample - Superstore.csv",encoding = "ISO-8859-1")
pd.set_option('display.max_colwidth', None)
df.head()

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                                  Product Name     Sales  \
0                            Bush Somerset Collection Bookcase  261.9600   
1  Hon Deluxe Fabric Upholstered Stacking Chairs, Rounded Back  731.9400   
2    Self-Adhesive Address Labels for Typewriters by Universal   14.6200   
3                Bretford CR4500 Series Slim Rectangular Table  957.5775   
4                               Eldon Fold 'N Roll Cart System   22.3680   

   Quantity  Discount    Profit  
0         2      0.00   41.9136  
1         3      0.00  219.5820  
2         2      0.00    6.8714  
3         5      0.45 -383.0310  
4         2      0.20    2.5164  

[5 rows x 21 columns]

In [3]:
df.shape

(9994, 21)

In [4]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [5]:
df.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [7]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [8]:
df.describe(include="object")

Order ID Order Date   Ship Date       Ship Mode Customer ID  \
count             9994       9994        9994            9994        9994   
unique            5009       1237        1334               4         793   
top     CA-2017-100111   9/5/2016  12/16/2015  Standard Class    WB-21850   
freq                14         38          35            5968          37   

        Customer Name   Segment        Country           City       State  \
count            9994      9994           9994           9994        9994   
unique            793         3              1            531          49   
top     William Brown  Consumer  United States  New York City  California   
freq               37      5191           9994            915        2001   

       Region       Product ID         Category Sub-Category     Product Name  
count    9994             9994             9994         9994             9994  
unique      4             1862                3           17             1850  
top      West  OFF-PA-10001970  Office Supplies      Binders  Staple envelope  
freq     3203               19             6026         1523               48

In [9]:
df[["Order Date","Ship Date"]].dtypes

Order Date    object
Ship Date     object
dtype: object

In [10]:
df["Order Date"]=pd.to_datetime(df["Order Date"])
df["Ship Date"]=pd.to_datetime(df["Ship Date"])

# CHECKING THE TOP SELLING PRODUCTS IN THE SUPERSTORE

In [11]:
top5_selling_products=df.groupby(["Product Name"])["Sales"].sum().sort_values(ascending=False).reset_index().head()
top5_selling_products

Product Name  \
0                                        Canon imageCLASS 2200 Advanced Copier   
1  Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind   
2                        Cisco TelePresence System EX90 Videoconferencing Unit   
3                                 HON 5400 Series Task Chairs for Big and Tall   
4                                   GBC DocuBind TL300 Electric Binding System   

       Sales  
0  61599.824  
1  27453.384  
2  22638.480  
3  21870.576  
4  19823.479

In [12]:

pio.renderers.default = 'notebook_connected'


In [13]:
fig=px.bar(top5_selling_products,x=top5_selling_products["Product Name"],y="Sales",
template="plotly_dark",
color_discrete_sequence= [["#A7C1A8","#D1D8BE","#819A91","#73946B","#6F826A"]],
title="Top 5 selling products in super store".upper(),
text_auto=True)
iplot(fig)

# CHECKING THE TOP PROFITABLE PRODUCTS IN THE SUPERSTORE

In [14]:
top5_profitable_products=df.groupby(["Product Name"])["Profit"].sum().sort_values(ascending=False).reset_index().head()
top5_profitable_products

Product Name  \
0                                        Canon imageCLASS 2200 Advanced Copier   
1  Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind   
2                                         Hewlett Packard LaserJet 3310 Copier   
3                                           Canon PC1060 Personal Laser Copier   
4                    HP Designjet T520 Inkjet Large Format Printer - 24" Color   

       Profit  
0  25199.9280  
1   7753.0390  
2   6983.8836  
3   4570.9347  
4   4094.9766

In [15]:
fig = px.bar(top5_profitable_products,
             x ="Profit" ,
             y="Product Name",
             template='plotly_dark',
             color="Profit",
             title='the top profitable products in the superstore '.upper() ,
             text_auto=True)

fig.update_layout(xaxis_title="Total Profit")


iplot(fig)


#### top5_selling_products:

#### • Canon imageCLASS 2200 Advanced Copier

#### • Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind

#### • Cisco Telepresence System EX90 Videoconferencing Unit

#### • HON 5400 Series Task Chairs for Big and Tall

#### • GBC DocuBind TL300 Electric Binding System

#### top5_profitable_products:

#### • Canon imageCLASS 2200 Advanced Copier

#### • Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind

#### • Hewlett Packard LaserJet 3310 Copier

#### • Canon PC1060 Personal Laser Copier

#### • HP Designjet T520 Inkjet Large Format Printer - 24" Color

# SALES TREND OVER TIME (MONTHLY,YEARLY)

In [16]:
df["Year"]=df["Order Date"].dt.year
df["Month"]=df["Order Date"].dt.to_period("M").astype(str)


In [17]:
df.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

        Product ID         Category Sub-Category  \
0  FUR-BO-10001798        Furniture    Bookcases   
1  FUR-CH-10000454        Furniture       Chairs   
2  OFF-LA-10000240  Office Supplies       Labels   
3  FUR-TA-10000577        Furniture       Tables   
4  OFF-ST-10000760  Office Supplies      Storage   

                                                  Product Name     Sales  \
0                            Bush Somerset Collection Bookcase  261.9600   
1  Hon Deluxe Fabric Upholstered Stacking Chairs, Rounded Back  731.9400   
2    Self-Adhesive Address Labels for Typewriters by Universal   14.6200   
3                Bretford CR4500 Series Slim Rectangular Table  957.5775   
4                               Eldon Fold 'N Roll Cart System   22.3680   

  Quantity Discount    Profit  Year    Month  
0        2     0.00   41.9136  2016  2016-11  
1        3     0.00  219.5820  2016  2016-11  
2        2     0.00    6.8714  2016  2016-06  
3        5     0.45 -383.0310  2015  2015-10  
4        2     0.20    2.5164  2015  2015-10  

[5 rows x 23 columns]

In [18]:
monthly_sales=df.groupby("Month")["Sales"].sum().reset_index()
yearly_sales=df.groupby("Year")["Sales"].sum().reset_index()


In [19]:
fig=px.bar(yearly_sales,
           x="Year",
           y="Sales",
           template="plotly_dark",
           color_discrete_sequence=["#94AF9F"],
           text_auto=True,
           title='Yearly Sales Trend')


iplot(fig)

In [20]:
fig=px.line(monthly_sales,
            x="Month",
            y="Sales",
            template="plotly_dark",
            color_discrete_sequence=["#B08BBB"],
            title="Monthly Sales Trend")
iplot(fig)

# WHICH CATEGORY OF PRODUCTS GENERATES THE HIGHEST PROFIT?

In [21]:
category=df.groupby(["Category"])["Profit"].sum().sort_values(ascending=False).reset_index()
category

Category       Profit
0       Technology  145454.9481
1  Office Supplies  122490.8008
2        Furniture   18451.2728

In [22]:
fig=px.bar(category,
           y="Category",
           x="Profit",
           template="plotly_dark",
           color_discrete_sequence=["#9EA1D4"],
           title="Total Profit For Each Category",
           text_auto=True)
fig.update_layout(bargap=0.6)

fig.show()

In [23]:
fig=px.pie(category,
           names="Category",
           values="Profit",
           template="plotly_dark",
           color_discrete_sequence=["#648DB3","#5459AC","#7965C1"],
           title="Profit Percentage for Each Category",
           hole=0.4)
         
fig.show()

# WHICH REGION GENERATES THE MORE SALES?

In [24]:
region_most_sales=df.groupby("Region")["Sales"].sum().sort_values(ascending=False).reset_index()
region_most_sales

Region        Sales
0     West  725457.8245
1     East  678781.2400
2  Central  501239.8908
3    South  391721.9050

In [25]:
fig=px.bar(region_most_sales,
           y="Region",
           x="Sales",
           template="plotly_dark",
           color_discrete_sequence=[["#AFDDFF","#A4CCD9","#8DBCC7","#8DD8FF"]],
           title="Most Sales in each Region",
           text_auto=True)
fig.update_layout(bargap=0.6)
fig.show()

# THE IMPACT OF DISCOUNTS ON SALES

In [26]:
fig=px.scatter(df,
               x="Discount",
               y="Sales",
               template="plotly_dark",
               color_discrete_sequence=["#67AE6E"],
               title="Total Sales with Discount",)
fig.show()

In [27]:
discount_group=df.groupby(["Discount"])["Profit"].sum().reset_index()
discount_group

Discount       Profit
0       0.00  320987.6032
1       0.10    9029.1770
2       0.15    1418.9915
3       0.20   90337.3060
4       0.30  -10369.2774
5       0.32   -2391.1377
6       0.40  -23057.0504
7       0.45   -2493.1111
8       0.50  -20506.4281
9       0.60   -5944.6552
10      0.70  -40075.3569
11      0.80  -30539.0392

In [28]:
fig=px.scatter(discount_group,
               x="Discount",
               y="Profit",
               template="plotly_dark",
               color_discrete_sequence=["#67AE6E"],
               title="Total Profit with Discount")
fig.show()

#### The best sales are when the discount is 50%, but it is not profitable.
#### The discount from 0% to 20% are having the more sales and getting best profits.

# THE AVERAGE PROFIT MARGIN FOR EACH PRODUCT CATEGORY

In [29]:
df["Profit Margin"]=df["Profit"]/df["Sales"]
df.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

          Category  Sub-Category  \
0        Furniture     Bookcases   
1        Furniture        Chairs   
2  Office Supplies        Labels   
3        Furniture        Tables   
4  Office Supplies       Storage   

                                                  Product Name     Sales  \
0                            Bush Somerset Collection Bookcase  261.9600   
1  Hon Deluxe Fabric Upholstered Stacking Chairs, Rounded Back  731.9400   
2    Self-Adhesive Address Labels for Typewriters by Universal   14.6200   
3                Bretford CR4500 Series Slim Rectangular Table  957.5775   
4                               Eldon Fold 'N Roll Cart System   22.3680   

  Quantity Discount    Profit  Year    Month  Profit Margin  
0        2     0.00   41.9136  2016  2016-11         0.1600  
1        3     0.00  219.5820  2016  2016-11         0.3000  
2        2     0.00    6.8714  2016  2016-06         0.4700  
3        5     0.45 -383.0310  2015  2015-10        -0.4000  
4        2     0.20    2.5164  2015  2015-10         0.1125  

[5 rows x 24 columns]

In [30]:
profit_margin_avg=df.groupby(["Category"])["Profit Margin"].mean().reset_index()
profit_margin_avg["Profit Margin"]=round(profit_margin_avg["Profit Margin"],2)
profit_margin_avg

Category  Profit Margin
0        Furniture           0.04
1  Office Supplies           0.14
2       Technology           0.16

In [31]:
fig=px.bar(profit_margin_avg,
           y="Profit Margin",
           x="Category",
           template="plotly_dark",
           color_discrete_sequence=["#48A6A7"],
          title="Average Profit Margin for each Category",text_auto=True)
fig.update_layout(bargap=0.6)
fig.show()